mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py

```diff
@@ -0,0 +1,146 @@
+import os.path
+import json
+import pandas as pd
+import pytz
+from tzlocal import get_localzone
+from datetime import datetime
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import Resource
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from mindsdb_sql import parse_sql
+
+from mindsdb.utilities import log
+from mindsdb.integrations.handlers.google_fit_handler.google_fit_tables import GoogleFitTable
+from mindsdb.integrations.libs.api_handler import APIHandler,FuncParser
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+)
+
+SCOPES = ['https://www.googleapis.com/auth/fitness.activity.read']
+DATE_FORMAT = '%Y-%m-%d'
+class GoogleFitHandler(APIHandler):
+
+    def __init__(self, name: str = None, **kwargs):
+        super().__init__(name)
+        args = kwargs.get('connection_data', {})
+        self.connection_args = {}
+        #TODO: make sure the arguments can read a list from user input when the database is created, since "redirect_uris" is a list.
+        for k in ['client_id', 'project_id', 'auth_uri',
+                  'token_uri', 'auth_provider_x509_cert_url', 'client_secret']:
+            if k in args:
+                self.connection_args[k] = args[k]
+            self.connection_args['redirect_uris'] = ["http://localhost"]
+        
+        self.api = None
+        self.is_connected = False
+
+        aggregated_data = GoogleFitTable(self)
+        self._register_table('aggregated_data', aggregated_data)
+
+    def connect(self) -> Resource:
+        if self.is_connected is True and self.api:
+            return self.api
+        if len(self.connection_args) == 7:
```

Comment:
Do we create a new file from the user's provided parameters? If so, maybe we can add an option for the user just to send one parameter, which will be the credentials.json file

Reply:
Hey! So if we add an option to just accept a credentials.json file, where would we want the user to put it? Do you think it is better to put it under mindsdb's topmost directory or within the Google Fit handler's directory, same as where the file would be generated now?

Reply:
@2170chm You can check the Google Big query example

*    [defining](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py#L156)
*    [using](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py#L38)

Reply:
@ZoranPandovski Scott and I have add the option to use credentials.json file. We have also fixed a few bugs and tested on our end.